File: backend/routers/stats.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter()


class StatsSummaryResponse(BaseModel):
    average_score: float
    most_improved_metric: str
    consecutive_days: int


@router.get("/summary", response_model=StatsSummaryResponse)
async def get_summary() -> StatsSummaryResponse:
    return StatsSummaryResponse(
        average_score=0.65,
        most_improved_metric="具体性",
        consecutive_days=3
    )


@router.get("/score-history")
async def get_score_history() -> dict:
    return {"message": "stub: get_score_history"}


@router.get("/category-distribution")
async def get_category_distribution() -> dict:
    return {"message": "stub: get_category_distribution"}

File: frontend/src/api/stats.ts
import { apiClient } from './client';
import type { StatsSummaryResponse } from '../types';

/**
 * GET /stats/summary
 * 成長ステータスのサマリーを取得する
 */
export const getStatsSummary = async (): Promise<StatsSummaryResponse> => {
    const response = await apiClient.get<StatsSummaryResponse>('/stats/summary');
    return response.data;
};

/**
 * GET /stats/score-history
 * 長期スコア推移を取得する
 */
export const getScoreHistory = async (): Promise<any[]> => {
    const response = await apiClient.get<any[]>('/stats/score-history');
    return response.data;
};

/**
 * GET /stats/category-distribution
 * テーマカテゴリ分布を取得する
 */
export const getCategoryDistribution = async (): Promise<any[]> => {
    const response = await apiClient.get<any[]>('/stats/category-distribution');
    return response.data;
};